src/services/goToDefinition.ts

```diff
@@ -108,24 +108,41 @@ namespace ts.GoToDefinition {
             || (!isCallLikeExpression(calledDeclaration.parent) && s === calledDeclaration.parent.symbol);
     }
 
-    export function getReferenceAtPosition(sourceFile: SourceFile, position: number, program: Program): { reference: FileReference, file: SourceFile } | undefined {
+    export function getReferenceAtPosition(sourceFile: SourceFile, position: number, program: Program): { reference: FileReference, fileName: string, file?: SourceFile } | undefined {
         const referencePath = findReferenceInPosition(sourceFile.referencedFiles, position);
         if (referencePath) {
             const file = program.getSourceFileFromReference(sourceFile, referencePath);
-            return file && { reference: referencePath, file };
+            return file && { reference: referencePath, fileName: file.fileName };
         }
 
         const typeReferenceDirective = findReferenceInPosition(sourceFile.typeReferenceDirectives, position);
         if (typeReferenceDirective) {
             const reference = program.getResolvedTypeReferenceDirectives().get(typeReferenceDirective.fileName);
             const file = reference && program.getSourceFile(reference.resolvedFileName!); // TODO:GH#18217
-            return file && { reference: typeReferenceDirective, file };
+            return file && { reference: typeReferenceDirective, fileName: file.fileName };
         }
 
         const libReferenceDirective = findReferenceInPosition(sourceFile.libReferenceDirectives, position);
         if (libReferenceDirective) {
             const file = program.getLibFileFromReference(libReferenceDirective);
-            return file && { reference: libReferenceDirective, file };
+            return file && { reference: libReferenceDirective, fileName: file.fileName };
+        }
+
+        if (sourceFile.resolvedModules?.size) {
+            const node = getTokenAtPosition(sourceFile, position);
+            if (isModuleSpecifierLike(node) && !pathIsBareSpecifier(node.text) && sourceFile.resolvedModules.has(node.text)) {
+                const fileName = sourceFile.resolvedModules.get(node.text)?.resolvedFileName

```

Comment:
i dont think we should do that.. what with option of using `rootDirs` to determine where this is relative to etc ?

Reply:
I assume you’re not talking about this line, but the next?

Reply:
The relative path resolution will only kick in if _real_ module resolution failed. This applies only to file types we don’t understand, like CSS, and to files that don’t exist, so you have a quick shortcut to create it. So yes, it’s a best guess, but I think it’s a pretty good guess (I don’t think very many people use `rootDirs`), and it replaces a total refusal to try.

Reply:
Yeah i meant next line... 
 wonder if these should be flagged as best guess or some such flag instead..
as @DanielRosenwasser pointed, we dont resolve imports currently in partial semantic mode, so adding a test that it works for that would be good too..

Reply:
Still believe we probably would want to flag guesses in definition so vscode can potentially show some kind of message or something like it does for completions ?

Reply:
@DanielRosenwasser @mjbvz thoughts? We can discuss at the editor meeting tomorrow if needed.